superbench/benchmarks/micro_benchmarks/directx_gemm_flops_performance/GPUCore.h

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#ifndef WIN32_LEAN_AND_MEAN
+#define WIN32_LEAN_AND_MEAN // Exclude rarely-used stuff from Windows headers.
+#endif
+
+#include <chrono>
+#include <string>
+#include <unordered_map>
+#include <wrl.h>
+#include <DirectXPackedVector.h>
+#include <d3d12.h>
+#include <d3d12shader.h>
+#include <d3dcompiler.h>
+#include <directml.h>
+#include <dxgi1_6.h>
+#include <windowsx.h>
+
+// linker
+#pragma comment(lib, "dxguid.lib")
+#pragma comment(lib, "dxgi.lib")
+#pragma comment(lib, "D3D12.lib")
+#pragma comment(lib, "d3dcompiler.lib")
+
+#define DEBUG 0
+#define _PRINT_RESULT 0
+
+#if defined(DEBUG)
+#include <dxgidebug.h>
+#endif
+
+#include "../directx_third_party/DXSampleHelper.h"
+#include "../directx_third_party/d3dx12.h"
+#include "../directx_utils/D3D12Timer.h"
+#include "Options.h"
+
+using namespace std;
+using namespace DirectX;
+// Note that while ComPtr is used to manage the lifetime of resources on the CPU,
+// it has no understanding of the lifetime of resources on the GPU. Apps must account
+// for the GPU lifetime of resources to avoid destroying objects that may still be
+// referenced by the GPU.
+// An example of this can be found in the class method: OnDestroy().
+using Microsoft::WRL::ComPtr;
+
+#define SAFE_RELEASE(p)                                                                                                \
+    if (p)                                                                                                             \
+    (p)->Release()
+template <typename T> T *get_rvalue_ptr(T &&v) { return &v; }
+
+class GPUCore {
+  public:
+    GPUCore(Options *opts) : opts(opts) {}
+    ~GPUCore() { FlushCommandQueue(); }
+
+    /**
+     * @brief Setup GPU and start benchmark.
+     */
+    void Run();
+
+    /**
+     * @brief Create pipeline including
+     *		  create device object, command list, command queue
+     *		  and synchronization objects.
+     */
+    void LoadPipeline();
+
+    /**
+     * @brief Prepare input and output data and buffers of the tensor elements..
+     */
+    template <typename T> void PrepareData(const int m, const int n, const int k);
+
+    /**
+     * @brief Setup and compile DirectML operator.
+     */
+    void LoadAssets(int m, int n, int k, DML_TENSOR_DATA_TYPE dataType);
+
+    /**
+     * @brief Initialize and tart the computation job.
+     * @return the elapsed time in ms.
+     */
+    template <typename T> double InitializeExecuteComputeOp(const int m, const int n, const int k);
```

Comment:
>const [](http://example.com/codeflow?start=60&length=5)

for a constant number, you don't need const, but it is OK to keep it. :smile:

Reply:
ok, got it